mlflow/sklearn/__init__.py

```diff
@@ -720,7 +726,26 @@ def _log_posttraining_metadata(estimator, *args, **kwargs):
             else:
                 try_mlflow_log(mlflow.log_metric, "training_score", training_score)
 
-        try_mlflow_log(log_model, estimator, artifact_path="model")
+
+        input_example = None
+        signature = None
+        if hasattr(estimator, "predict"):
```

Comment:
@harupy I simplified the logic here to encapsulate the entire row sampling + signature inference procedure into a single try/catch block. While the error messages may be a bit less granular, the key point users should be aware of is that the input example and signature are not logged in the event of an error; this structure makes that a bit clearer, and it improves the readability of the code.